test/e2e/apimachinery/aggregator.go

```diff
@@ -570,3 +577,59 @@ func generateFlunderName(base string) string {
 	}
 	return fmt.Sprintf("%s-%d", base, id)
 }
+
+func waitAPICoherentResponse(f *framework.Framework) {
+	framework.Logf("Waiting on v1alpha1.wardle.example.com to disappear from all apiserver instances")
+	client := f.ClientSet
+	namespace := f.Namespace.Name
+	timeout := 300
+
+	// kubectl create -f role.yaml
+	_, err := client.RbacV1().Roles("default").Create(context.TODO(), &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "endpoint-getter",
+		},
+		Rules: []rbacv1.PolicyRule{
+			rbacv1helpers.NewRule("get").Groups("").Resources("endpoints").RuleOrDie(),
+		},
+	}, metav1.CreateOptions{})
+	framework.ExpectNoError(err, "creating role %s in namespace default", "endpoint-getter")
+	defer func(){
+		err = client.RbacV1().Roles("default").Delete(context.TODO(), "endpoint-getter", metav1.DeleteOptions{})
+		framework.ExpectNoError(err, "deleting role %s in namespace default", "endpoint-getter")
+	}()
+
+	_, err = client.RbacV1().RoleBindings("default").Create(context.TODO(), &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "agnhost:" + namespace + ":endpoint-getter",
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     "endpoint-getter",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      "default",
+				Namespace: namespace,
+			},
+		},
+	}, metav1.CreateOptions{})
+	framework.ExpectNoError(err, "creating role binding %s in namespace default", "agnhost:"+namespace+":endpoint-getter")
+	defer func(){
+		err = client.RbacV1().RoleBindings("default").Delete(context.TODO(), "agnhost:"+namespace+":endpoint-getter", metav1.DeleteOptions{})
+		framework.ExpectNoError(err, "deleting role binding %s in namespace default", "agnhost:"+namespace+":endpoint-getter")
+	}()
+
+	// Prepare the agnostic host pod to watch for a coherent answer from all API servers. Expect not finding the wardle group name and version.
+	podClient := f.PodClient()
+	agnPod := e2epod.NewAgnhostPod(namespace, "agnhost", nil, nil, nil, "api-coherent", "--api=wardle.example.com/v1alpha1", "--status=404", fmt.Sprintf("--timeout=%d", timeout))
```

Comment:
I think we can do this creating a pod and executing directly on it with some curl and bash glue

Reply:
Is this a common pattern? Was following @liggitt advice from https://github.com/kubernetes/kubernetes/pull/100585 about creating a new command in `agnhost`. Is it preferrable to have scripts over go commands in tests?

Reply:
is not about using scripts on pods, agree that is not preferably, but the way I see this, is a simple curl execution, you don't need a script, the logic happens in the e2e test , you just execute the command on the pod and get the output, the logic remains the same